core/src/test/java/org/apache/calcite/sql/test/AbstractSqlTester.java

```diff
@@ -307,15 +317,17 @@ public SqlTester withOperatorTable(SqlOperatorTable operatorTable) {
     return with("operatorTable", operatorTable);
   }
 
-  public SqlTesterImpl withConnectionFactory(
+  public SqlTester withConnectionFactory(
       CalciteAssert.ConnectionFactory connectionFactory) {
     return with("connectionFactory", connectionFactory);
   }
 
-  protected SqlTesterImpl with(final String name, final Object value) {
-    return new SqlTesterImpl(factory.with(name, value));
+  protected SqlTester with(final String name, final Object value) {
+    return createImpl(factory.with(name, value));
   }
 
+  protected abstract SqlTester createImpl(SqlTestFactory factory);
```

Comment:
Could you please name consistent naming? `SqlTester with(SqlTestFactory factory)`